source/server/server.cc

```diff
@@ -646,6 +646,15 @@ absl::Status InstanceBase::initializeOrThrow(Network::Address::InstanceConstShar
   admin_ = std::make_shared<AdminImpl>(initial_config.admin().profilePath(), *this,
                                        initial_config.admin().ignoreGlobalConnLimit());
 
+  auto typed_admin_impl_ = dynamic_cast<AdminImpl*>(admin_.get());
```

Comment:
hmm not sure if we want this. would you be able to remove the use of dynamic_cast? we in generally avoid using it.